Add language property support

jeffreyvanhees opened the pull request:
The Mapbox Geocoding API supports a [language property](https://docs.mapbox.com/api/search/geocoding/#forward-geocoding-with-search-text-input) to return results in the appropriate language. This simple PR adds support for that.

Dutch (`nl`):

```php
Geocoder::make('address')
 ->label('Adres')
 ->language('nl')
```
<img width="315" height="201" alt="Screenshot 2025-10-15 at 14 12 05" src="https://github.com/user-attachments/assets/3b6cce3b-0f90-4436-a0dc-3473f7a7e8a3" />

jeffreyvanhees committed:
Update geocoder.blade.php

jeffreyvanhees committed:
Update Geocoder.php

jeffreyvanhees committed:
Update README.md

jeffreyvanhees committed:
Set default language to `en`

jeffreyvanhees commented:
@speniti Right now I’m setting the default language to English, but it might be an option to allow the property to be null so that it automatically picks up the client’s browser language. I think there should probably be an @if / @endif around it in the Blade view (not sure), but I’ll leave that up to you. :)

speniti commented:
Hey @jeffreyvanhees, thanks so much for the PR! 
I'd love to suggest changing the default to `null`, and maybe even omitting the option so that we can inherit the document language. 
I can think of one situation where this option might come in handy. Sometimes, you might want to use a language different from the one indicated in the `<html>` tag. 
I understand it might seem like more of a rare case, but I think it's a good option to have.

jeffreyvanhees commented:
@speniti agree and on it! Will update the PR soon :)

jeffreyvanhees committed:
Set $language to null as default

jeffreyvanhees committed:
Filter out all properties that are `null` in blade view

jeffreyvanhees commented:
@speniti I updated the PR with the following:

- Default language is now `null`
- Updated the blade view to filter out all properties that are `null` or an empty array/string, so it will keep the Mapbox-defaults:

```php
<div
  x-load
  x-load-src="{{ FilamentAsset::getAlpineComponentSrc('geocoder', 'speniti/filament-mapbox') }}"
  x-data="geocoder($wire.{{ $applyStateBindingModifiers("\$entangle('{$getStatePath()}')") }},
  @js(array_filter([
    'accessToken' => $getAccessToken(),
    'clearAndBlurOnEsc' => $getClearAndBlurOnEsc(),
    'countries' => $getCountries(),
    'language' => $getLanguage(),
    'fuzzyMatch' => $getFuzzyMatch(),
    'limit' => $getLimit(),
    'minLength' => $getMinLength(),
    'placeholder' => $getPlaceholder(),
    'types' => $getTypes(),
  ], fn($value) => !is_null($value) && $value !== '' && $value !== []), JSON_THROW_ON_ERROR),
  @js($isDisabled(), JSON_THROW_ON_ERROR))"
  x-ignore
  wire:ignore
></div>
```

speniti commented:
Hey @jeffreyvanhees! I tried to merge your PR, but there are a couple of checks that didn't pass. I'd be happy to wait for you to address those issues so we can proceed with the merge.
Before pushing your changes, I would recommend running the `composer lint` script to check for resolution. 😉

jeffreyvanhees committed:
Lint fixes

jeffreyvanhees committed:
Fix tests
